src/pages/index.js

```diff
@@ -4,52 +4,142 @@ import FormValidator from "../components/FormValidator.js";
-  initialCards,
   addCardButton,
+  profileAvatarContainer,
   profileButtonEdit,
-  profileModalInputName,
-  profileModalInputSubtitle,
-  profileModalForm,
 } from "../utils/constants.js";
 import { config } from "../utils/constants.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
+import Api from "../components/Api.js";
+import ConfirmDeletePopup from "../components/ConfDeletePopup.js";
+import ProfilePicPopup from "../components/ProfilePicPopup.js";
 
-// Create the instances of the classes
-const imagePopup = new PopupWithImage("#imageOpen");
+// Api Initialization
+export const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "fbd43f39-bc6f-4ff1-b85c-3a8e63a0b02d",
+    "Content-Type": "application/json",
+  },
+});
+
+// CARDS AND PAGE RENDERING
+// Getting the user data and the initial cards on the initial page loading
+
+let cardsList; // правильно ли так??? Было раньше как внизу, теперь CardsList объявлена так, чтобы использоваться в попапе
+
+// api
+//   .loadPageContent()
+//   .then(([initialCards, userData]) => {
+//     // здесь возвращается 2 промиса, и нужно каждый обработать
+//     // Рендеринг карточек снаружи api ?????
+//     const cardsList = new Section(
+//       {
+//         items: initialCards,
+//         renderer: (item) => {
+//           const newCard = createCard(item);
+//           cardsList.addItem(newCard);
+//         },
+//       },
+//       config.cardsListSelector
+//     );
+
+//     userInfo.setUserInfo({
+//       // разобраться детально, почему так
+//       name: userData.name,
+//       description: userData.about,
+//       avatar: userData.avatar,
+//     });
+//     cardsList.renderItems();
+//   })
+//   .catch((err) => console.log(err));
+
+api
+  .loadPageContent()
+  .then(([initialCards, userData]) => {
+    // Устанавливаем информацию о пользователе
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+
+    // Инициализируем экземпляр Section и сохраняем его в глобальной переменной cardsList
+    cardsList = new Section(
+      {
+        items: initialCards,
+        renderer: (item) => {
+          const newCard = createCard(item);
+          cardsList.addItem(newCard);
+        },
+      },
+      config.cardsListSelector
+    );
+
+    cardsList.renderItems();
+  })
+  .catch((err) => console.error(err));
+
+// Card creation function
 function createCard(item) {
-  const cardElement = new Card(item, "#card-template", () =>
-    imagePopup.open(item)
+  const cardElement = new Card(
+    item,
+    "#card-template",
+    () => imagePopup.open(item),
+    (isLiked) => {
+      if (isLiked) {
+        api.dislikeCard(item._id).then((res) => {
+          cardElement.updateLikeStatus(res.isLiked);
+        });
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**